Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIStorage.mm

```diff
@@ -66,12 +71,123 @@
         @"browser.test.assertThrows(() => browser.storage.local.remove({}), /'keys' value is invalid, because a string or an array of strings is expected, but an object was provided/i)",
         @"browser.test.assertThrows(() => browser.storage.local.remove([1]), /'keys' value is invalid, because a string or an array of strings is expected, but an array of other values was provided/i)",
 
+        @"browser.test.assertThrows(() => browser.storage.session.setAccessLevel('INVALID_ACCESS_LEVEL'), /'accessOptions' value is invalid, because an object is expected/i)",
+        @"browser.test.assertThrows(() => browser.storage.session.setAccessLevel({ 'accessLevel': 'INVALID_ACCESS_LEVEL' }), /'accessLevel' value is invalid, because it must specify either 'TRUSTED_CONTEXTS' or 'TRUSTED_AND_UNTRUSTED_CONTEXTS'/i)",
+
         @"browser.test.notifyPass()"
     ]);
 
     Util::loadAndRunExtension(storageManifest, @{ @"background.js": backgroundScript });
 }
 
+TEST(WKWebExtensionAPIStorage, UndefinedProperties)
+{
+    auto *backgroundScript = Util::constructScript(@[
+        @"browser.test.assertEq(typeof browser.storage.local.setAccessLevel, 'undefined')",
+        @"browser.test.assertEq(typeof browser.storage.sync.setAccessLevel, 'undefined')",
+
+        @"browser.test.assertEq(typeof browser.storage.local.QUOTA_BYTES_PER_ITEM, 'undefined')",
+        @"browser.test.assertEq(typeof browser.storage.session.QUOTA_BYTES_PER_ITEM, 'undefined')",
+
+        @"browser.test.assertEq(typeof browser.storage.local.MAX_ITEMS, 'undefined')",
+        @"browser.test.assertEq(typeof browser.storage.session.MAX_ITEMS, 'undefined')",
+
+        @"browser.test.assertEq(typeof browser.storage.local.MAX_WRITE_OPERATIONS_PER_HOUR, 'undefined')",
+        @"browser.test.assertEq(typeof browser.storage.session.MAX_WRITE_OPERATIONS_PER_HOUR, 'undefined')",
+
+        @"browser.test.assertEq(typeof browser.storage.local.MAX_WRITE_OPERATIONS_PER_MINUTE, 'undefined')",
+        @"browser.test.assertEq(typeof browser.storage.session.MAX_WRITE_OPERATIONS_PER_MINUTE, 'undefined')",
```

Comment:
This is fine as-is. But could be:

```
browser.test.assertEq(browser.storage.local.setAccessLevel, undefined)
```

Also we should start using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) for property access. That will prevent the test from throwing an exception if any of the properties in the "chain" are null / undefined. Especially since an exception causes our tests to timeout currently.

```
browser.test.assertEq(browser?.storage?.local?.setAccessLevel, undefined)
```